buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -79,10 +75,7 @@ class PluginBuildPlugin implements Plugin<Project> {
             project.extensions.getByType(PluginPropertiesExtension).extendedPlugins.each { pluginName ->
                 // Auto add dependent modules to the test cluster
                 if (project.findProject(":modules:${pluginName}") != null) {
-                    project.integTest.dependsOn(project.project(":modules:${pluginName}").tasks.bundlePlugin)
-                    project.testClusters.integTest.module(
-                            project.project(":modules:${pluginName}").tasks.bundlePlugin.archiveFile
-                    )
+                    project.testClusters.integTest.module(":modules:${pluginName}")
```

Comment:
Building on what Jake has been doing, I think we should make this `project.testClusters.all { }`, such that regardless of what kind of test plugin is applied (yaml rest, java rest, etc) we auto wire this up for folks, rather than relying on the loose convention of `integTest`. 

Additionally, I think he has some stuff in flight that removes this task entirely and makes all this stuff explicit.